akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorTimersSpec.scala

```diff
@@ -0,0 +1,132 @@
+
+import java.util.UUID
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.concurrent.duration._
+
+import akka.actor.testkit.typed.scaladsl._
+import akka.actor.typed.ActorRef
+import akka.actor.typed.Behavior
+import akka.actor.typed.scaladsl.Behaviors
+import akka.persistence.typed.PersistenceId
+import akka.testkit.EventFilter
+import akka.testkit.TestEvent.Mute
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import org.scalatest.WordSpecLike
+
+object EventSourcedBehaviorTimersSpec {
+
+  val journalId = "event-sourced-behavior-timers-spec"
+
+  def config: Config = ConfigFactory.parseString(s"""
+        akka.loglevel = INFO
+        akka.loggers = [akka.testkit.TestEventListener]
+        akka.persistence.journal.leveldb.dir = "target/typed-persistence-${UUID.randomUUID().toString}"
+        akka.persistence.journal.plugin = "akka.persistence.journal.leveldb"
+        """)
+
+  def testBehavior(persistenceId: PersistenceId, probe: ActorRef[String]): Behavior[String] =
+    Behaviors.setup { _ =>
+      Behaviors.withTimers { timers =>
+        EventSourcedBehavior[String, String, String](
+          persistenceId,
+          emptyState = "",
+          commandHandler = (_, command) =>
+            command match {
+              case "scheduled" =>
+                probe ! "scheduled"
+                Effect.none
+              case "cmd-0" =>
+                timers.startSingleTimer("key", "scheduled", Duration.Zero)
+                Effect.none
+              case _ =>
+                timers.startSingleTimer("key", "scheduled", Duration.Zero)
+                Effect.persist(command).thenRun(_ => probe ! command)
+            },
+          eventHandler = (state, evt) => state + evt)
+      }
+    }
+
+  def testTimerFromSetupBehavior(persistenceId: PersistenceId, probe: ActorRef[String]): Behavior[String] =
+    Behaviors.setup { _ =>
+      Behaviors.withTimers { timers =>
+        timers.startSingleTimer("key", "scheduled", Duration.Zero)
+
+        EventSourcedBehavior[String, String, String](
+          persistenceId,
+          emptyState = "",
+          commandHandler = (_, command) =>
+            command match {
+              case "scheduled" =>
+                probe ! "scheduled"
+                Effect.none
+              case _ =>
+                Effect.persist(command).thenRun(_ => probe ! command)
+            },
+          eventHandler = (state, evt) => state + evt)
+      }
+    }
+
+}
+
+class EventSourcedBehaviorTimersSpec
+    extends ScalaTestWithActorTestKit(EventSourcedBehaviorTimersSpec.config)
+    with WordSpecLike {
+
+  import EventSourcedBehaviorTimersSpec._
+
+  val pidCounter = new AtomicInteger(0)
+  private def nextPid(): PersistenceId = PersistenceId(s"c${pidCounter.incrementAndGet()})")
+
+  import akka.actor.typed.scaladsl.adapter._
+  // needed for the untyped event filter
+  private implicit val untypedSystem: akka.actor.ActorSystem = system.toUntyped
```

Comment:
add the issue number we have in the other places now

Reply:
ok, I'll do that in https://github.com/akka/akka/pull/26579
although it's all over the place when we have the real event filter in place we should search for all usages of the old one